servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/RetryingHttpRequesterFilter.java

```diff
@@ -842,6 +872,22 @@ public Builder retryOther(
             return this;
         }
 
+        /**
+         * Callback invoked on every {@link StreamingHttpClient#request(StreamingHttpRequest) request} retry attempt.
+         * <p>
+         * This can be used to track when {@link BackOffPolicy} actually decides to retry a request, to update
+         * {@link HttpRequestMetaData request meta-data} before a retry, or implement logging/metrics.
+         *
+         * @param onRequestRetry {@link BiConsumer} with {@link HttpRequestMetaData request meta-data} and
+         * {@link Throwable cause} that is invoked before every
+         * {@link StreamingHttpClient#request(StreamingHttpRequest) request} retry attempt
+         * @return {@code this}
+         */
+        public Builder onRequestRetry(final BiConsumer<HttpRequestMetaData, Throwable> onRequestRetry) {
```

Comment:
Would it be useful if the number of retries on the current attempt are also made available to the consumer somehow? I know one can do this manually by setting it on the context and tracking it on their own, but I would assume that the number of attempts is something a lot of people need, if only for logging.

Reply:
Good point. I will have to come up with a new interface. Will do that, thanks!

Reply:
Any suggestion for the name? Assuming we may need a similar feature for `reserveConnection` retries

Reply:
Its not very apparent from the name at which stage the flow runs other than the javadoc. WDYT about including such information as part of the name? Something like `onBeforeRequestRetry`

Reply:
Regarding the interface my 2c would be `BeforeRetryConsumer` or `BeforeRetryInfluencer` to highlight the ability to change things in the flow.

Reply:
I will probably go with `beforeRequestRetry(BeforeRetryCallback)` if no objections

Reply:
I ended up keeping the builder name as-is, but clarifying "before" as part of the `RetryCallbacks` contract.

Naming is hard, open to other suggestions in the context of a new interface.